### DISCUSSION TITLE
How to fit the multi-band data using the phoebe?

### DISCUSSION THREAD

**Discussion by AstroJLin** [phoebe-project/phoebe2#589] [Questions]
Hi Kyle,

In fact, I want to fit the multi-band data using the phoebe as done  in Saracino et al. (2022). Here,  should Saracino et al. (2022) fit the I and V band together using the model of the PHOEBE to derive the parameters of the model and determine uncertainties with emcee?  However,  I only able to fit respectively the single band using my model including to determine uncertainties with emcee.

So I would like to know how to set up to achieve this result. 

Thanks!

The result is as follow (Saracino et al. 2022):
[1.pdf](https://github.com/phoebe-project/phoebe2/files/8631628/1.pdf)
[2.pdf](https://github.com/phoebe-project/phoebe2/files/8631638/2.pdf)

**Comment by gecheline** [comment]
Hi AstroJLin,

To fit multiple light curves (or even multiple types of datasets) with emcee in PHOEBE, all you need to do is add them to the bundle and ensure they are enabled in the compute options you use for emcee (they should be all enabled by default). So, if you have two separate light curves, just add two separate datasets for them. Here's a basic example:

```
lc_V = np.loadtxt('my_V_light_curve.txt')
lc_I = np.loadtxt('my_I_light_curve.txt')

b = phoebe.default_binary()
b.add_dataset('lc', times=lc_V[:,0], fluxes=lc_V[:,1], sigmas=lc_V[:,2], passband='Johnson:V', dataset='lcV')
b.add_dataset('lc', times=lc_I[:,0], fluxes=lc_I[:,1], sigmas=lc_I[:,2], passband='Johnson:I', dataset='lcI')
```

After you have these data sets in the bundle, you can enable or disable each one with b.enable_dataset() and b.disable_dataset(). If you always want to fit both, there's no need to change anything - once you add them to the bundle, each call to run_compute() and run_solver() will take both of those light curves into consideration and will fit your model to both of them within emcee.

Hope this helps answer your question. If you are running into any specific problems/errors with your system, let me know!

**Comment by AstroJLin** [comment]
Hi gecheline,

I have been able to take both of those light curves into consideration and fit my model to both of them within emcee. 

Here, I wonder how to plot those light curves (multi-band data) fit from emcee within a picture as done Saracino et al. 2022 (2.pdf).

In addition, I have nine free parameters in my model. I should how to set up the niters, progress_every_niters and nwalkers for the emcee  so that it can convergence and run fast.

Thanks!

**Reply by gecheline** [reply]
Hi AstroJLin,

For plotting the corner plot and resulting light curves from the fit, see the following example: http://phoebe-project.org/docs/2.3/examples/inverse_paper_examples . The section "Accessing posteriors from emcee run" covers how to make the plot in 2.pdf and the section "Propagating Posteriors through Forward-Model" covers plotting the model with uncertainties like in 1.pdf.
